tutorials/nservicebus-101/lesson-1/tutorial.md

```diff
@@ -0,0 +1,144 @@
+
+```no-highlight
+Install-Package NServiceBus -ProjectName ClientUI
+```
+
+This adds a reference to the NServiceBus.Core assembly to the project. With the proper dependencies in place, we're ready to start writing code.
+
+
+### Configure an endpoint
+
+Although NServiceBus is an async framework, the console application's `Main()` method is not. Therefore, it's necessary to create an entry point into code where we can use the `async`/`await` keywords.
+
+In the **Program.cs** file, modify the code to look like the following:
+
+snippet:EmptyProgram
+
+Now we're ready to create a [**messaging endpoint**](/nservicebus/endpoints/). A messaging endpoint (or just **endpoint**) is a logical component that's capable of sending and receiving messages. An endpoint is hosted within a process, which in this case is a simple console application, but could be a web application or other .NET process.
+
+Let's take a look at all the code required to run an endpoint within a console application as a block, and afterward we'll analyze it line-by-line.
+
+Add this code to your AsyncMain method:
+
+snippet:AsyncMain
+
+Let's go line-by-line and find out exactly what each step is doing.
+
+
+#### Console Title
+
+snippet:ConsoleTitle
+
+When developing NServiceBus systems, we tend to run multiple message endpoints at the same time in separate console windows. If we're not careful, it can become difficult to tell them apart. Giving each one a short, clear name can be helpful to make it easier to find the desired window later on.
+
+In later lessons, we'll expand our project to have several endpoints, and this one will be the test endpoint we'll use to send messages to the rest of the system. Thus we give it the name `ClientUI`.
+
+
+#### EndpointConfiguration
+
+snippet:EndpointName
+
+The `EndpointConfiguration` class is where we define all the settings that determine how our endpoint will operate. The single string parameter `ClientUI` is the [**endpoint name**](/nservicebus/endpoints/specify-endpoint-name.md), which serves as the logical identity for our endpoint, and forms a naming convention by which other things will derive their names, such as the **input queue** where the endpoint will listen for messages to process.
+
+
+#### Message transport
+
+snippet:Transport
+
+This setting defines the [**message transport**](/nservicebus/transports/) that NServiceBus will use to send and receive messages. `MsmqTransport` is the only transport available within the core library. All other transports require additional NuGet packages.
+
+The [**MSMQ transport**](/nservicebus/msmq/) is the default setting, so we technically don't need this line at all. However, if you add a period right before the semicolon (in order to see the IntelliSense) you will see that more settings related to the message transport are available as extension methods. We will explore these in later lessons. For now, it's good to make the choice of transport explicit within our code.
```

Comment:
Colloquial is what I'm going for, and it is necessary later on when we do the MSMQ-specific routing stuff. We didn't make it required in V6 but I want to give the impression that you should have it all the time, and hint at the configuration options that are hidden behind it by having them peek at the intellisense.